### PR TITLE
v4.0 Raven db 11497 40 well known

### PR DESCRIPTION
**Pull request by ml054**

### CODE REVIEW

File: src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs [hunk 1]
```diff
@@ -525,6 +527,8 @@ public Task GetCertificates()
                         writer.WriteComma();
                         writer.WritePropertyName("LoadedServerCert");
                         writer.WriteString(Server.Certificate.Certificate?.Thumbprint);
+                        writer.WriteComma();
+                        writer.WriteArray("WellKnownCerts", wellKnown);
```
Comment (ppekrol) [inline src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs]: I would change that to `WellKnownAdminCerts`

File: src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs [hunk 2]
```diff
@@ -554,35 +558,53 @@ public Task WhoAmI()
...
+                        }
+                    }
+                    else if (wellKnown != null && wellKnown.Contains(clientCert.Thumbprint, StringComparer.OrdinalIgnoreCase))
                     {
-                        // Since we didn't go through EnsureNotPassive(), the server certificate is not registered yet so we'll add it to the local state.
                         var serverCertDef = new CertificateDefinition
                         {
-                            Name = "Server Certificate",
-                            Certificate = Convert.ToBase64String(Server.Certificate.Certificate.Export(X509ContentType.Cert)),
+                            Name = "Well Known Certificate",
```
Comment (ppekrol) [inline src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs]: `Well Known Admin Certificate`

File: src/Raven.Studio/wwwroot/App/views/manage/certificates.html [hunk 3]
```diff
@@ -53,7 +53,37 @@ <h2 class="text-warning"><i class="icon-unsecure"></i> Authentication is off</h2
...
+            <div class="scroll flex-grow">
+                <div class="panel panel-info certificate-item" data-bind="visible: wellKnownCerts().length">
+                    <div class="padding padding-sm flex-horizontal">
+                        <div class="certificate-name">
+                            <div class="flex-horizontal">
+                                <div>
+                                    <i class="icon-certificate"></i>    
+                                </div>
+                                <div>
+                                    <h3 class="no-margin">Well known certificates defined by system administrator</h3>
```
Comment (ppekrol) [inline src/Raven.Studio/wwwroot/App/views/manage/certificates.html]: Well known admin certificates...

File: src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs (hunk 1 above)
Comment (ml054) [reply src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs]: done

File: src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs (hunk 2 above)
Comment (ml054) [reply src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs]: done

File: src/Raven.Studio/wwwroot/App/views/manage/certificates.html (hunk 3 above)
Comment (ml054) [reply src/Raven.Studio/wwwroot/App/views/manage/certificates.html]: done

**Comment by ml054** [comment]
@ppekrol please review again

File: src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs [hunk 4]
```diff
@@ -554,35 +558,53 @@ public Task WhoAmI()
...
 
-                    if (certificate == null && Server.Certificate.Certificate != null)
+                    if (clientCert.Equals(Server.Certificate.Certificate))
+                    {
+                        var certKey = Constants.Certificates.Prefix + clientCert.Thumbprint;
+                        using (ctx.OpenReadTransaction())
+                            certificate = ServerStore.Cluster.Read(ctx, certKey) ??
+                                          ServerStore.Cluster.GetLocalState(ctx, certKey);
+
+                        if (certificate == null && Server.Certificate.Certificate != null)
```
Comment (ayende) [inline src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs]: Why do you have this here? Why add the server cert to the local store?

**Review by ayende — CHANGES_REQUESTED** [review]

**Pull request merged**